pkg/app/api/grpcapi/piped_api.go

```diff
@@ -679,6 +681,55 @@ func (a *PipedAPI) ReportApplicationLiveStateEvents(ctx context.Context, req *pi
 	return &pipedservice.ReportApplicationLiveStateEventsResponse{}, nil
 }
 
+// GetLatestEvent gives back the latest event that meets the given conditions.
+func (a *PipedAPI) GetLatestEvent(ctx context.Context, req *pipedservice.GetLatestEventRequest) (*pipedservice.GetLatestEventResponse, error) {
```

Comment:
I'm still debating letting it give back a list like `ListNotCompletedDeployments`. Honestly not sure it's able to perform IN query against map[string]string for all Datastore type well. I attempted to put off it but seems like I'd better carefully look into it at this point.